core/blockchain.go

```diff
@@ -916,33 +916,29 @@ func (bc *BlockChain) WriteBlockWithState(block *types.Block, receipts []*types.
 		bc.triegc.Push(root, -float32(block.NumberU64()))
 
 		if current := block.NumberU64(); current > triesInMemory {
+			// If we exceeded our memory allowance, flush matured singleton nodes to disk
+			var (
+				nodes, imgs = triedb.Size()
+				limit       = common.StorageSize(bc.cacheConfig.TrieNodeLimit) * 1024 * 1024
+			)
+			if nodes > limit || imgs > 4*1024*1024 {
```

Comment:
What's the `4*1024*1024` limit for `imgs`, and why is that not a named parameter like the other settings?

Reply:
This is a random number really, just needs to be small enough not to be bothersome, large enough to dedup data. It's arbitrary really.